src/components/pages/settings/TeamsForm.js

```diff
@@ -75,26 +76,27 @@ class TeamsForm extends React.Component {
 
   onAreaChange = (selected) => {
     this.setState({selectedAreas: selected});
-    this.form.areas = selected.split(',');
+    const updatedArea = selected.split(',');
+    if (this.form.areas !== updatedArea) this.form.areas = updatedArea;
   }
 
   handleFormUpdate = (field, value) => {
-    this.form[field] = value;
+    if (this.form[field] !== value) this.form[field] = value;
```

Comment:
This is still a mutation, to prevent mutations instead of reassigning values you'll need to replace the whole object with the old values and the updated ones.
A good way to achieve this is by using `Object.assign` (don't forget the empty object as the first param, or you'll incurr in mutations):
```this.form = Object.assign({}, this.form, { [field]: value }); ```
A better/safer way is using as a shorthand the spread operator (this will always create a new object): 
``` this.form = { ...this.form, [field]: value }; ```
This way a copy of the old object is made and you don't have to worry about mutations.
**More info:** https://slemgrim.com/mutate-or-not-to-mutate/